File: row-change-event-channel/src/main/java/nl/haploid/octowight/AppConfiguration.java
package nl.haploid.octowight;

import nl.haploid.octowight.JsonMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
import org.springframework.scheduling.annotation.EnableScheduling;

@Configuration
@EnableScheduling
public class AppConfiguration {

	@Bean
	public static PropertySourcesPlaceholderConfigurer propertyPlaceholderConfigurer() {
		return new PropertySourcesPlaceholderConfigurer();
	}

	@Bean
	public JsonMapper jsonMapper() {
		return new JsonMapper();
	}
}

File: event-emitter/src/main/scala/nl/haploid/octowight/configuration/AppConfiguration.scala
package nl.haploid.octowight.configuration

import nl.haploid.octowight.JsonMapper
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
import org.springframework.scheduling.annotation.EnableScheduling

object AppConfiguration {

  @Bean def propertyPlaceholderConfigurer: PropertySourcesPlaceholderConfigurer = new PropertySourcesPlaceholderConfigurer
}

@Configuration
@EnableScheduling
class AppConfiguration {

  @Bean def jsonMapper: JsonMapper = new JsonMapper
}
